### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by panianchuk**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const copyState = { ...state };
+
+  for (const action of actions) {
```
Comment (Bohdan-Maksymiuk) [inline src/transformStateWithClones.js:13]: also use [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) for usage of type, extraData, keysToRemove

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,27 @@
...
+      for (const i of action.keysToRemove) {
+        if (copyState.hasOwnProperty(i)) {
+          delete copyState[i];
+        }
+      }
+    } else if (action.type === 'clear') {
+      for (const i in copyState) {
+        delete copyState[i];
+      }
+    }
```
Comment (Bohdan-Maksymiuk) [inline src/transformStateWithClones.js:26]: avoid such nesting `else if` conditions
This code is hard to read and maintain

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,35 @@
...
-  // write code here
+  const result = [];
+  const copyState = { ...state };
+
+  for (const action of actions) {
+    const { keysToRemove, extraData, type } = action;
+
+    switch (true) {
+      case type === 'addProperties':
+        Object.assign(copyState, extraData);
```
Comment (Bohdan-Maksymiuk) [inline src/transformStateWithClones.js:18]:
```suggestion
    switch (type) {
      case 'addProperties':
        Object.assign(copyState, extraData);
```
Inappropriate usage of switch

Comment (Bohdan-Maksymiuk) [reply src/transformStateWithClones.js:18]: [UL](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch)

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

**Review by Viktor-Kost — APPROVED** [review]

**Pull request closed without merging**